### PR TITLE
sonic-buildimage: Arista-7060X6-64DE-O128S2

### PR DESCRIPTION
**Pull request by ccroy-arista**
#### Why I did it
Baseline implementation for 256x100g support in the Arista-7060X6-64PE (DCS-7060X6-64PE) has been requested by MSFT.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Added the necessary files for baseline implementation of 128x400g implementation for Quicksilver QSFP-DD.

#### How to verify it
Apply change to DUT and confirm that the expected interfaces are up.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305
- [x] 202405

#### Tested branch (Please provide the tested image version)
202405

#### Description for the changelog
Baseline support for the Arista-7060X6-64DE-O128S2 platform variant.

### CODE REVIEW

File: device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm [hunk 1]
```diff
@@ -0,0 +1,1416 @@
+#
+# $Copyright: (c) 2022 Broadcom.
+# Broadcom Proprietary and Confidential. All rights reserved.$
+#
+# BCM78900 64x800g port configuration.
```
Comment (r12f) [inline device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm:5]: hi Chris, I didn't see any UC/MC queue settings as well as buffer initialization in the BCM file, will the switch still be able to perform the L2/L3 forwarding with this configuration?

Comment (davidm-arista) [reply device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm:5]: Are you seeing any issues with L2/L3 forwarding?

Comment (r12f) [reply device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm:5]: yea, David. The L2 MC would not work from what we see in Tristan. Let's get this in first. We can create another BCM case to get the BCM config.

**Review by r12f — APPROVED** [review]

File: device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm [hunk 2]
```diff
@@ -0,0 +1,1416 @@
...
+                ENABLE: 0
+                MAX_FRAME_SIZE: 9416
+                SPEED: 10000
+                NUM_LANES: 1
+...
+---
+bcm_device:
+    0:
+        global:
+            ftem_mem_entries: 65536
```
Comment (kperumalbfn) [inline device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm:1397]: @ccroy-arista Can you check the other BCM file - https://github.com/sonic-net/sonic-buildimage/blob/master/device/arista/x86_64-arista_7060x6_64pe/Arista-7060X6-64PE-256x200G/th5-a7060x6-64pe.config.bcm

Some of the flexcounter configs are missing here.

File: device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm [hunk 3]
```diff
@@ -0,0 +1,1416 @@
...
+...
+---
+device:
+    0:
+        # Per pipe flex counter configuration
+        CTR_EFLEX_CONFIG:
```
Comment (kperumalbfn) [inline device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm:1403]: Flex counter should be pipemode to support stats for all the queue/ports.

File: device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm (hunk 2 above)
Comment (ccroy-arista) [reply device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm:1397]: Thanks @kperumalbfn, flex counter configs added.

File: device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm (hunk 3 above)
Comment (ccroy-arista) [reply device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm:1412]: Pipemode updated.

Comment (kperumalbfn) [reply device/arista/x86_64-arista_7060x6_64de/Arista-7060X6-64DE-O128S2/th5-a7060x6-64de.config.bcm:1412]: Thanks @ccroy-arista

**Review by kperumalbfn — APPROVED** [review]

**Comment by StormLiangMS** [comment]
hi @bingwang-ms would you help to cherry pick? This is for TH5.

**Pull request merged**